src/graph/transform/cuda/cuda_to_block.cu

```diff
@@ -174,13 +177,16 @@ struct CUDAIdsMapper {
           src_nodes, rhs_nodes, &node_maps, count_lhs_device, &lhs_nodes,
           stream);
 
-      device->CopyDataFromTo(
-          count_lhs_device, 0, num_nodes_per_type.data(), 0,
-          sizeof(*num_nodes_per_type.data()) * num_ntypes, ctx,
-          DGLContext{kDGLCPU, 0}, DGLDataType{kDGLInt, 64, 1});
-      device->StreamSync(ctx, stream);
+      CUDA_CALL(cudaEventCreate(&copyEvent));
+      new_len_tensor = NDArray::PinnedEmpty(
+          {num_ntypes}, DGLDataTypeTraits<int64_t>::dtype,
+          DGLContext{kDGLCPU, 0});
+      CUDA_CALL(cudaMemcpyAsync(
+          new_len_tensor->data, count_lhs_device,
+          sizeof(*num_nodes_per_type.data()) * num_ntypes,
+          cudaMemcpyDeviceToHost, stream));
+      CUDA_CALL(cudaEventRecord(copyEvent, stream));
 
-      // Wait for the node counts to finish transferring.
       device->FreeWorkspace(ctx, count_lhs_device);
```

Comment:
Is there a risk that `count_lhs_device` will be freed before the copying is finished?

Reply:
As the copy is happening on the current cuda stream retrieved by `runtime::getCurrentCUDAStream()`, this allocation should not be available for re-use until the copy operation finishes.